### PR TITLE
Add compatibility with Symfony 4

### LINKED ISSUE #49: Support to Symfony Process ~4.0
Thanks for this package.

Any plans for adding support to symfony/process ~4.0?

### PR DESCRIPTION
**Pull request by pyrech**
Fix #49

Also drop support for Symfony < 3.3.
Drop support for PHP < 7 & HHVM.
Add typehint everywhere.
Better CI and tests configuration.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Changes between versions
 
+## Not yet released
```
Comment (lyrixx) [inline CHANGELOG.md]: `## 2.0.0 (not released yet)` ?

There are many BC break, IMHO, it should be 2.0

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+cs:
```
Comment (lyrixx) [inline Makefile]: What about adding doc: https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html ?

File: src/Notifier/CliBasedNotifier.php [hunk 3]
```diff
@@ -59,13 +59,13 @@ public function isSupported()
...
+    public function send(Notification $notification): bool
     {
         if (!$notification->getBody()) {
             throw new InvalidNotificationException($notification, 'Notification body can not be empty');
         }
 
-        $builder = new ProcessBuilder();
+        $arguments = $this->getCommandLineArguments($notification);
 
         if (self::SUPPORT_BINARY_PROVIDED === $this->support && $this instanceof BinaryProvider) {
```
Comment (lyrixx) [inline src/Notifier/CliBasedNotifier.php]: Oh, this not how OOP should work ;)

File: CHANGELOG.md (hunk 1 above)
Comment (pyrech) [reply CHANGELOG.md]: I'm not sure if Notifier interface was part of the public API

Comment (pyrech) [reply CHANGELOG.md]: Anyway, I can bump to 2.0, it won't hurt anyone ^^

File: Makefile (hunk 2 above)
Comment (pyrech) [reply Makefile]: Hum, why not

File: src/Notifier/CliBasedNotifier.php (hunk 3 above)
Comment (pyrech) [reply src/Notifier/CliBasedNotifier.php]: Indeed, this is a bit ugly but it works for now. Feel free to open a PR to fix that 😉

Comment (lyrixx) [reply src/Notifier/CliBasedNotifier.php]: I'm fine with that ;) LazyMe

**Pull request merged**